web/src/main/java/org/openmrs/web/filter/GZIPFilter.java

```diff
@@ -34,13 +35,15 @@
 	
 	private Boolean cachedGZipEnabledFlag = null;
 	
+	private String cachedGZipCompressedRequestForPathAccepted = null;
+	
 	/**
 	 * @see org.springframework.web.filter.OncePerRequestFilter#doFilterInternal(javax.servlet.http.HttpServletRequest,
 	 *      javax.servlet.http.HttpServletResponse, javax.servlet.FilterChain)
 	 */
 	public void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain chain)
 	        throws IOException, ServletException {
-		
+		request = performGZIPRequest(request);
```

Comment:
You must catch the exception catch(APIException e) { response.sendError(HttpServletResponse.SC_UNSUPPORTED_MEDIA_TYPE); return; }